tests/utils/stl/test/format.py

```diff
@@ -31,7 +32,31 @@ class TestStep:
     file_deps: List[os.PathLike] = field(default_factory=list)
     env: Dict[str, str] = field(default_factory=dict)
     should_fail: bool = field(default=False)
+    executor: Executor = field(default=None)
+    def run(self):
+        return self.executor.run(self.cmd, self.work_dir,
+                                 self.file_deps, self.env)
 
+@dataclass
+class SigningRunStep(TestStep):
+    is_kernel: bool = field(default=False)
+    wdk_bin: str = field(default=None)
+
+    cert_path: os.PathLike = field(default=Path(''))
+    cert_pass: str = field(default="")
+    def run(self):
+        # TODO clean up this hacky mess
+        if self.is_kernel:
+            # sign the binary before running it.
+            cmd = [self.wdk_bin + '/x64/signtool.exe', 'sign',
+                   '/f', self.cert_path,
+                   '/p', self.cert_pass,
+                   self.cmd[0]]
+            out, err, rc = stl.util.executeCommand(cmd, self.work_dir)
+            if rc != 0:
+                self.should_fail = False
+                return (cmd, out, err, rc)
+        return super().run()

```

Comment:
As the comment suggests, I don't think too highly of this approach.  A more correct approach may be to add a build step enum that covers signing, or to make the link step also do signing.